plugins/devcontainer-github-codespaces/src/index.ts

```diff
@@ -0,0 +1,101 @@
+import type {
+  AmplicationPlugin,
+  CreateAdminUIParams,
+  CreateServerParams,
+  DsgContext,
+  EventParams,
+  Events,
+  ModuleMap,
+} from "@amplication/code-gen-types";
+import { EventNames } from "@amplication/code-gen-types";
+import { getPluginSettings } from "./utils";
+import genDevcontainerConfig from "./utils/genDevcontainerConfig"
+import genDevcontainerConfigWithAdminUI from "./utils/genDevcontainerConfigWithAdminUI"
+import patchNginx from "./utils/patchNginx"
+import { readFile } from "fs/promises";
+import path, { join } from "path"
+import fs from "fs"
+import { Context } from "vm";
+import Module from "module";
+import genPathBasedOnConfig from "./utils/genPathBasedOnConfig"
+import { Settings } from "./types";
+
+class GithubCodespacesPlugin implements AmplicationPlugin {
+  register(): Events {
+    return {
+      [EventNames.CreateServer]: {
+        after: this.afterCreateServer,
+      },
+      [EventNames.CreateAdminUI]: {
+        after: this.afterCreateAdminUI
+      },
+    };
+  }
+
+  async afterCreateServer(
+    context: DsgContext,
+    eventParams: CreateServerParams,
+    modules: ModuleMap
+  ): Promise<ModuleMap> {
+    const settings = getPluginSettings(context.pluginInstallations);
+    const serviceName = context.resourceInfo?.name ?? "Amplication App"
+    const devContainerPath = genPathBasedOnConfig(settings, serviceName)
+
+    // Get dev container config
+    var containerConfig;
+    if (!settings.includeAdminUI) {
+      containerConfig = genDevcontainerConfig(serviceName, context.serverDirectories.baseDirectory)
+    } else {
+      containerConfig = genDevcontainerConfigWithAdminUI(serviceName, context.serverDirectories.baseDirectory, context.clientDirectories.baseDirectory)
+    }
+
+    const envConfig = fs.readFileSync(join(__dirname, "templates", "init.sh"), { encoding: "utf-8" })
+      .replace("{SERVER_ROOT}", context.serverDirectories.baseDirectory)
+      .replace("{CLIENT_ROOT}", context.clientDirectories.baseDirectory)
+
+    // Merge code
+    await modules.set({
+      code: JSON.stringify(containerConfig),
```

Comment:
You can use prettier to prettify the code also.

![image](https://github.com/amplication/plugins/assets/100484401/75b1c312-83d4-4ec2-9229-1b979d49b325)

Reply:
Example:- https://github.com/amplication/plugins/blob/e8b38999d71c29303e856a269acf77a6522e3957/plugins/formatter-prettier/src/index.ts#L63